orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/dialect/impl/OrientDBEntityQueries.java

```diff
@@ -0,0 +1,209 @@
+ */
+package org.hibernate.datastore.ogm.orientdb.dialect.impl;
+
+import com.orientechnologies.orient.core.id.ORecordId;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import org.hibernate.datastore.ogm.orientdb.constant.OrientDBConstant;
+import org.hibernate.datastore.ogm.orientdb.logging.impl.Log;
+import org.hibernate.datastore.ogm.orientdb.logging.impl.LoggerFactory;
+import org.hibernate.datastore.ogm.orientdb.utils.EntityKeyUtil;
+import org.hibernate.ogm.dialect.spi.AssociationContext;
+import org.hibernate.ogm.model.key.spi.AssociationKey;
+import org.hibernate.ogm.model.key.spi.EntityKey;
+import org.hibernate.ogm.model.key.spi.EntityKeyMetadata;
+import org.hibernate.ogm.model.spi.Tuple;
+import org.hibernate.ogm.model.impl.DefaultEntityKeyMetadata;
+
+/**
+ * @author Sergey Chernolyas (sergey.chernolyas@gmail.com)
+ */
+public class OrientDBEntityQueries extends QueriesBase {
+
+	private static Log LOG = LoggerFactory.getLogger();
+
+	private final EntityKeyMetadata entityKeyMetadata;
+
+	public OrientDBEntityQueries(EntityKeyMetadata entityKeyMetadata) {
+		this.entityKeyMetadata = entityKeyMetadata;
+		for ( int i = 0; i < entityKeyMetadata.getColumnNames().length; i++ ) {
+			String columnName = entityKeyMetadata.getColumnNames()[i];
+			LOG.info( "column number:" + i + "; column name:" + columnName );
+		}
+
+	}
+
+	/**
+	 * Find the node corresponding to an entity.
+	 *
+	 * @param executionEngine the {@link GraphDatabaseService} used to run the query
+	 * @param columnValues the values in {@link org.hibernate.ogm.model.key.spi.EntityKey#getColumnValues()}
+	 * @return the corresponding node
+	 * @throws java.sql.SQLException
+	 */
+
+	public Map<String, Object> findEntity(Connection executionEngine, EntityKey entityKey) throws SQLException {
+		Map<String, Object> params = params( entityKey.getColumnValues() );
+		Map<String, Object> dbValues = new LinkedHashMap<>();
+		Object dbKeyValue = EntityKeyUtil.findPrimaryKeyValue( entityKey );
+		String dbKeyName = EntityKeyUtil.findPrimaryKeyName( entityKey );
+
+		Statement stmt = executionEngine.createStatement();
+		StringBuilder query = new StringBuilder( "select from " );
+		if ( params.size() == 1 ) {
+			if ( dbKeyValue instanceof ORecordId ) {
+				// search by @rid
+				ORecordId rid = (ORecordId) dbKeyValue;
+				query.append( rid );
+			}
+			else {
+				// search by business key
+				query.append( entityKey.getTable() ).append( " WHERE " ).append( dbKeyName ).append( " = " );
+				EntityKeyUtil.setFieldValue( query, dbKeyValue );
+			}
+		}
+		else {
+			// query.append(entityKeyMetadata.getTable());
+			throw new UnsupportedOperationException( "Not supported yet." );
```

Comment:
It would be nice if you could add a more descriptive message, what's not supported?